libs/template/class/src/lib/test/class.directive.spec.ts

```diff
@@ -0,0 +1,166 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  CUSTOM_ELEMENTS_SCHEMA
+} from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { BehaviorSubject, of } from 'rxjs';
+import { ClassDirective } from '../class.directive';
+
+@Component({
+  selector: 'rx-test-component',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div
+      class='container-one'
+      [rxClass]='containerOneClasses$'
+      [rxClassStrategy]="'native'"
+    ></div>
+    <div
+      class='container-two'
+      [rxClass]='containerTwoClasses$'
+      [rxClassStrategy]="'native'"
+    ></div>
+    <div
+      class='container-three'
+      [rxClass]="{
+        'class-one': setClassOne$,
+        'class-two': setClassTwo$
+      }"
+      [rxClassStrategy]="'native'"
+    ></div>
+    <div
+      class='container-four'
+      [rxClass]='containerFourClasses$'
+      [rxClassStrategy]="'native'"
+    ></div>
+    <div
+      class='container-five'
+      [rxClass]='containerFiveClasses$'
+      [rxClassStrategy]="'native'"
+    ></div>
+    <div
+      class='container-six'
+      [rxClass]='containerSixClass$'
+      [rxClassStrategy]="'native'"
+    ></div>
+  `
+})
+class TestComponent {
+  readonly containerOneClasses$ = of({
+    'class-one': true,
+    'class-two': true
+  });
+  readonly containerTwoClasses$ = of(['class-one', 'class-two']);
+  readonly containerFourClasses$ = new BehaviorSubject<string[]>([
+    'class-one',
+    'class-two'
+  ]);
+  readonly containerFiveClasses$ = new BehaviorSubject<Set<string>>(
+    new Set(['class-one', 'class-two'])
+  );
+  readonly containerSixClass$ = new BehaviorSubject<string>('class-one');
+  readonly setClassOne$ = of(true);
+  readonly setClassTwo$ = of(false);
+}
+
+describe('rxClass', () => {
+  let fixture: ComponentFixture<TestComponent>;
+
+  beforeEach(() => {
+    fixture = TestBed.configureTestingModule({
+      declarations: [TestComponent, ClassDirective],
+      schemas: [CUSTOM_ELEMENTS_SCHEMA]
```

Comment:
isn't it easier to just provide the primary strategy as native here instead of setting native for each directive instance?